### DISCUSSION TITLE
Single screen for editing all "Publish Layouts" Role allocations

### DISCUSSION THREAD

**Discussion by ic360-com** [ExpressionEngine/ExpressionEngine#4213] [Feature Requests]
### Suggest an idea for ExpressionEngine

When creating a new Role, it can be a pain to edit all the Publish Layouts, to get the new Role allocated - especially if there are 20+ channels.

It would be awesome to have a single screen to achieve this, listing all the Channels, their Publish Layouts and the Roles... with Checkboxes / Radio Buttons (perhaps) to switch the required allocations on/off.

**Comment by shbchk** [comment]
++++

**Comment by intoeetive** [comment]
Perhaps not directly related, but I'm thinking of an Utility that would allow to set relationships "anything to anything".

You pick a model, then pick model instance (e.g. "Layouts" -> "%layout name%") it then populates the possible relationships that you can make based on model metadata (e.g. select "Roles"), in then populated the list of that model instances (e.g. Roles) and then you select the related models.

It might be more difficult to implement than it seems though, so just as idea for now

**Reply by ic360-com** [reply]
Sounds interesting. I'd be interested to see more... am trying to think in my head how this would be useful in real world situations too. More of a tool for the site builders I guess, rather than content editors? Although I wonder if it could benefit both.
